File: sources/Android/android-communications/library/src/main/java/com/polar/androidcommunications/api/ble/BleRefApiVersion.java
package com.polar.androidcommunications.api.ble;

import com.polar.androidcommunications.BuildConfig;

public class BleRefApiVersion {
    // just use simplified latest tag version
    public final static String VERSION_STRING = BuildConfig.GIT_VERSION;

    public static int major() {
        String[] components = BuildConfig.GIT_VERSION.split("\\.");
        if (components.length != 0) {
            return Integer.parseInt(components[0]);
        }
        return 0;
    }

    public static int minor() {
        String[] components = BuildConfig.GIT_VERSION.split("\\.");
        if (components.length > 1) {
            return Integer.parseInt(components[1]);
        }
        return 0;
    }

    public static int patch() {
        String[] components = BuildConfig.GIT_VERSION.split("\\.");
        if (components.length > 2) {
            return Integer.parseInt(components[2]);
        }
        return 0;
    }
}

File: sources/Android/android-communications/library/src/main/java/com/polar/androidcommunications/api/ble/BleRefApiVersion.kt
package com.polar.androidcommunications.api.ble

import com.polar.androidcommunications.BuildConfig

object BleRefApiVersion {
    // just use simplified latest tag version
    const val VERSION_STRING: String = BuildConfig.GIT_VERSION

    fun major(): Int {
        val components =
            BuildConfig.GIT_VERSION.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }
                .toTypedArray()
        if (components.size != 0) {
            return components[0].toInt()
        }
        return 0
    }

    fun minor(): Int {
        val components =
            BuildConfig.GIT_VERSION.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }
                .toTypedArray()
        if (components.size > 1) {
            return components[1].toInt()
        }
        return 0
    }

    fun patch(): Int {
        val components =
            BuildConfig.GIT_VERSION.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }
                .toTypedArray()
        if (components.size > 2) {
            return components[2].toInt()
        }
        return 0
    }
}
